python/ribasim_testmodels/ribasim_testmodels/basic.py

```diff
@@ -346,7 +346,7 @@ def tabulated_rating_curve_model() -> ribasim.Model:
     rating_curve = ribasim.TabulatedRatingCurve(
         static=pd.DataFrame(
             data={
-                "node_id": [2, 2],
+                "node_id": [2, 3],
```

Comment:
This doesn't seem right. Node 3 is in the time table.

Reply:
The static table is basically a collection of lookup tables stacked on top of each other. That is why multiple lines have the same node ID, because they describe one discharge/level relation. For node ID 2 this relation is fixed in time and described by two rows, and for node ID 3 the relation varies with time and is in the `time` table.

https://deltares.github.io/Ribasim/core/usage.html#tabulatedratingcurve

Reply:
It should also be noted that this went under the radar because the the `tabulated_rating_curve` testmodel does not in fact appear in any test ^^

Reply:
That model serves mainly to test updating tables over time. There is a test using that here: https://github.com/Deltares/Ribasim/blob/d208665dce98d35671f00f18d97636686b4e7a04/core/test/basin.jl#L27. Though without #258 it can still easily go under the radar.

> I assumed that every node type class has a mandatory static field with all node IDs for that node type appearing at least once

No, at least I didn't assume this so far. If you have a time series for some parameter and want to use that, it seems unnecessary extra work to ask users to provide a static value for that parameter even though it won't be used.

> Is there a way to obtain the node IDs from the node type objects without making a specific case for each node type?

Not really, no. So far I always use the node table for this:

https://github.com/Deltares/Ribasim/blob/d208665dce98d35671f00f18d97636686b4e7a04/core/src/io.jl#L6

Although I guess you could take the union of all `node_id` columns.